openedx_learning/apps/authoring/units/api.py

```diff
@@ -151,6 +152,7 @@ def create_next_unit_version(
         created=created,
         created_by=created_by,
         container_version_cls=UnitVersion,
+        entities_action=entities_action,
     )
     return unit_version
 
```

Comment:
Ugh.. it's inconsistent that we require a `Unit` object to call Unit API methods, when we require a `container_key` for the Container API methods.

I'd much rather we passed in a `container_key` for _all_ API calls, to prevent users of this API from having to fetch objects when they don't need to -- and I think it would be a simple refactor to get that happening here.

@ormsbee what do you think?

Reply:
Yes, containers `create_next_container_version` api expects [id](https://github.com/open-craft/openedx-learning/blob/7552d7a762707110d255034bd4fcd6c042f3c16d/openedx_learning/apps/authoring/publishing/api.py#L846) while `create_next_unit_version` expects unit object. It is probably due to this [statement](https://github.com/open-craft/openedx-learning/blob/7552d7a762707110d255034bd4fcd6c042f3c16d/openedx_learning/apps/authoring/publishing/api.py#L881) that selects related publishable_entities.

But it seems like `create_next_container_version` is low level API as it all its arguments are native types like str, int and datetime, while `create_next_unit_version` is a higher level api that wraps the former and accepts objects like `unit` and `components` instead of ids.

Reply:
> it's inconsistent that we require a Unit object to call Unit API methods, when we require a container_key for the Container API methods.

For what it's worth, we also kind of discussed this here on the previous PR: https://github.com/openedx/openedx-learning/pull/278#discussion_r1980520531